### PR TITLE
Curating Owners: pkg/controller

### PR DESCRIPTION
**Pull request by apelisse**
cc @jsafrane @mikedanese @bprashanth @derekwaynecarr @thockin @saad-ali

In an effort to expand the existing pool of reviewers and establish a
two-tiered review process (first someone **lgtms** and then someone
experienced in the project **approves**), we are adding new reviewers to
existing owners files.
## If You Care About the Process:

We did this by algorithmically figuring out who’s contributed code to
the project and in what directories.  Unfortunately, that doesn’t work
perfectly: people that have made mechanical code changes (e.g change the
copyright header across all directories) end up as reviewers in lots of
places.

Instead of using pure commit data, we generated an excessively large
list of reviewers and pruned based on all time commit data, recent
commit data and review data (number of PRs commented on).

At this point we have a decent list of reviewers, but it needs one last
pass for fine tuning.
## TLDR:

As an owner of a sig/directory and a leader of the project, here’s what
we need from you:
1. Use PR https://github.com/kubernetes/kubernetes/pull/35715 as an example.
2. The pull-request is made editable, please edit the OWNERS file to add
   the names of people that should be reviewing code in the future in the **reviewers** section. You probably do NOT need to modify the **approvers** section.
3. Notify me if you want some OWNERS file to be removed.  Being an approver or reviewer
   of a parent directory makes you a reviewer/approver of the subdirectories too, so not all
   OWNERS files may be necessary.
4. Please use ALIAS if you want to use the same list of people over and
   over again (don't hesitate to ask me for help, or use the pull-request
   above as an example)

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/35833)

### CODE REVIEW

**Comment by saad-ali** [comment]
Thanks @apelisse. I'll take care of the `volume/` subdir.

**Review by bprashanth — COMMENTED** [review]
This is very nice of you :) I would've taken the opposite approach, if your name is in git blame you know enough to delegate ownership. Will take a look tomorrow.

**Comment by derekwaynecarr** [comment]
Will review tomorrow

File: pkg/controller/daemon/OWNERS [hunk 1]
```diff
@@ -0,0 +1 @@
+reviewers:
```
Comment (bprashanth) [inline pkg/controller/daemon/OWNERS]: @mikedanese

File: pkg/controller/deployment/OWNERS [hunk 2]
```diff
@@ -0,0 +1 @@
+reviewers:
```
Comment (bprashanth) [inline pkg/controller/deployment/OWNERS]: @janetkuo 
@nikhiljindal 
@kargakis

File: pkg/controller/informers/OWNERS [hunk 3]
```diff
@@ -0,0 +1 @@
+reviewers:
```
Comment (bprashanth) [inline pkg/controller/informers/OWNERS]: @wojtek-t 
@lavalamp

File: pkg/controller/job/OWNERS [hunk 4]
```diff
@@ -0,0 +1 @@
+reviewers:
```
Comment (bprashanth) [inline pkg/controller/job/OWNERS]: @erictune

File: pkg/controller/namespace/OWNERS [hunk 5]
```diff
@@ -0,0 +1 @@
+reviewers:
```
Comment (bprashanth) [inline pkg/controller/namespace/OWNERS]: @derekwaynecarr

File: pkg/controller/node/OWNERS [hunk 6]
```diff
@@ -0,0 +1 @@
+reviewers:
```
Comment (bprashanth) [inline pkg/controller/node/OWNERS]: @gmarek

File: pkg/controller/petset/OWNERS [hunk 7]
```diff
@@ -0,0 +1 @@
+reviewers:
```
Comment (bprashanth) [inline pkg/controller/petset/OWNERS]: @foxish 
@bprashanth

File: pkg/controller/podautoscaler/OWNERS [hunk 8]
```diff
@@ -0,0 +1 @@
+reviewers:
```
Comment (bprashanth) [inline pkg/controller/podautoscaler/OWNERS]: @mwielgus ?

File: pkg/controller/replicaset/OWNERS [hunk 9]
```diff
@@ -0,0 +1 @@
+reviewers:
```
Comment (bprashanth) [inline pkg/controller/replicaset/OWNERS]: @kargakis 
@caesarxuchao 
@lavalamp
@bprashanth

File: pkg/controller/replication/OWNERS [hunk 10]
```diff
@@ -0,0 +1 @@
+reviewers:
```
Comment (bprashanth) [inline pkg/controller/replication/OWNERS]: @caesarxuchao 
@lavalamp
@kargakis 
@bprashanth

File: pkg/controller/route/OWNERS [hunk 11]
```diff
@@ -0,0 +1 @@
+reviewers:
```
Comment (bprashanth) [inline pkg/controller/route/OWNERS]: @cjcullen

File: pkg/controller/scheduledjob/OWNERS [hunk 12]
```diff
@@ -0,0 +1 @@
+reviewers:
```
Comment (bprashanth) [inline pkg/controller/scheduledjob/OWNERS]: @erictune 
@janetkuo

File: pkg/controller/service/OWNERS [hunk 13]
```diff
@@ -0,0 +1 @@
+reviewers:
```
Comment (bprashanth) [inline pkg/controller/service/OWNERS]: @bprashanth
@bowei 
@MrHohn

File: pkg/controller/endpoint/OWNERS [hunk 14]
```diff
@@ -0,0 +1 @@
+reviewers:
```
Comment (bprashanth) [inline pkg/controller/endpoint/OWNERS]: @bprashanth
@bowei 
@MrHohn

File: pkg/controller/garbagecollector/OWNERS [hunk 15]
```diff
@@ -0,0 +1 @@
+reviewers:
```
Comment (bprashanth) [inline pkg/controller/garbagecollector/OWNERS]: @caesarxuchao

File: pkg/controller/serviceaccount/OWNERS [hunk 16]
```diff
@@ -0,0 +1 @@
+reviewers:
```
Comment (bprashanth) [inline pkg/controller/serviceaccount/OWNERS]: @liggitt

File: pkg/controller/podgc/OWNERS [hunk 17]
```diff
@@ -0,0 +1 @@
+reviewers:
```
Comment (bprashanth) [inline pkg/controller/podgc/OWNERS]: @mikedanese

File: pkg/controller/resourcequota/OWNERS [hunk 18]
```diff
@@ -0,0 +1 @@
+reviewers:
```
Comment (bprashanth) [inline pkg/controller/resourcequota/OWNERS]: @derekwaynecarr

File: pkg/controller/certificates/OWNERS [hunk 19]
```diff
@@ -0,0 +1 @@
+reviewers:
```
Comment (bprashanth) [inline pkg/controller/certificates/OWNERS]: @deads2k 
@mikedanese

File: pkg/controller/disruption/OWNERS [hunk 20]
```diff
@@ -0,0 +1 @@
+reviewers:
```
Comment (bprashanth) [inline pkg/controller/disruption/OWNERS]: @mml 
@davidopp

**Review by bprashanth — COMMENTED** [review]
I've at'ted appropriate reviewers, please add yourself (by editing the file), delegate or nack by mid-week.

File: pkg/controller/job/OWNERS (hunk 4 above)
Comment (kargakis) [inline pkg/controller/job/OWNERS]: + @soltysh

File: pkg/controller/petset/OWNERS (hunk 7 above)
Comment (kargakis) [inline pkg/controller/petset/OWNERS]: @smarterclayton 
@kargakis

File: pkg/controller/scheduledjob/OWNERS (hunk 12 above)
Comment (kargakis) [inline pkg/controller/scheduledjob/OWNERS]: @soltysh

File: pkg/controller/daemon/OWNERS (hunk 1 above)
Comment (kargakis) [inline pkg/controller/daemon/OWNERS]: @kargakis
@lukaszo

File: pkg/controller/deployment/OWNERS (hunk 2 above)
Comment (kargakis) [inline pkg/controller/deployment/OWNERS]: @mfojtik

File: pkg/controller/informers/OWNERS (hunk 3 above)
Comment (kargakis) [inline pkg/controller/informers/OWNERS]: @deads2k

File: pkg/controller/podautoscaler/OWNERS (hunk 8 above)
Comment (kargakis) [inline pkg/controller/podautoscaler/OWNERS]: @jszczepkowski ?
@piosz ?

File: pkg/controller/petset/OWNERS (hunk 7 above)
Comment (janetkuo) [inline pkg/controller/petset/OWNERS]: @janetkuo

**Comment by apelisse** [comment]
@janetkuo @bprashanth @kargakis We probably need more names than that, please use the generated list here: https://github.com/kubernetes/kubernetes/pull/35833/commits/9ad2492d7666240b46eb151153a9b78fcd13e228 as a starting point.

File: pkg/controller/garbagecollector/OWNERS (hunk 15 above)
Comment (derekwaynecarr) [inline pkg/controller/garbagecollector/OWNERS]: @deads2k

File: pkg/controller/disruption/OWNERS (hunk 20 above)
Comment (gmarek) [inline pkg/controller/disruption/OWNERS]: @mwielgus

File: pkg/controller/route/OWNERS (hunk 11 above)
Comment (gmarek) [inline pkg/controller/route/OWNERS]: Not @cjcullen, I guess @gmarek or @wojtek-t

File: pkg/controller/node/OWNERS (hunk 6 above)
Comment (derekwaynecarr) [inline pkg/controller/node/OWNERS]: @ingvagabund @aveshagarwal would be good if you can get in the reviewer pool here.

File: pkg/controller/informers/OWNERS (hunk 3 above)
Comment (caesarxuchao) [inline pkg/controller/informers/OWNERS]: I want to be a reviewer as well.

**Comment by kargakis** [comment]
> @janetkuo @bprashanth @kargakis We probably need more names than that, please use the generated list here: 9ad2492 as a starting point.

I think people opting in or being added by others instead of using a generated list is better because many of the people in the list are not active for controller reviews anymore.

File: pkg/controller/podautoscaler/OWNERS (hunk 8 above)
Comment (derekwaynecarr) [inline pkg/controller/podautoscaler/OWNERS]: @DirectXMan12

File: pkg/controller/service/OWNERS (hunk 13 above)
Comment (derekwaynecarr) [inline pkg/controller/service/OWNERS]: @kubernetes/rh-networking -- any volunteers?

**Comment by apelisse** [comment]
@kargakis We use a generated list just as a suggestion to help you add people

**Comment by bprashanth** [comment]
The list has a lot of people who mostly won't care, I'd rather keep this to a small list of people who will? @apelisse

**Comment by apelisse** [comment]
Everyone, please feel free to modify the files directly. They are editable and create a new commit. That will make it much easier for me to keep track of.

Thanks!

File: pkg/controller/node/OWNERS (hunk 6 above)
Comment (smarterclayton) [inline pkg/controller/node/OWNERS]: @smarterclayton

File: pkg/controller/namespace/OWNERS (hunk 5 above)
Comment (smarterclayton) [inline pkg/controller/namespace/OWNERS]: @smarterclayton

File: pkg/controller/service/OWNERS (hunk 13 above)
Comment (bprashanth) [inline pkg/controller/service/OWNERS]: rh-networking please add volunteers to both service and endpoints controller, currently these just contain the networking team members here

File: pkg/controller/podautoscaler/OWNERS (hunk 8 above)
Comment (DirectXMan12) [inline pkg/controller/podautoscaler/OWNERS]: I'd recommend @fgrzadkowski as well

File: pkg/controller/serviceaccount/OWNERS (hunk 16 above)
Comment (liggitt) [inline pkg/controller/serviceaccount/OWNERS]: @liggitt as approver/reviewer is fine

File: pkg/controller/OWNERS [hunk 21]
```diff
@@ -1,4 +1,5 @@
-assignees:
-  - bprashanth
-  - derekwaynecarr
-  - mikedanese
+approvers:
+- bprashanth
+- derekwaynecarr
+- mikedanese
+reviewers:
```
Comment (deads2k) [inline pkg/controller/OWNERS]: > @deads2k here as someone who knows most of the controller patterns

sure.

**Comment by apelisse** [comment]
I've updated the list with comments posted on this PR, and squashed/grouped commits by author.

**Comment by mikedanese** [comment]
@apelisse You could probably squash to one commit. These commits are not very interesting.

**Comment by apelisse** [comment]
Yeah, I don't have a strong opinion. @pwittrock wanted to keep each commit for accountability I guess.

**Comment by apelisse** [comment]
@jsafrane @mikedanese @bprashanth @derekwaynecarr @thockin @saad-ali 

Looks like it could be merged now, one of you, feel free to LGTM.

**Review by bprashanth — APPROVED** [review]
LGTM, added a few reviewers but mostly looks ok
